### PR TITLE
[BIT] 转换能力补齐：scatter scatter_nd scatter_nd_add floor_mod frexp

### PR DESCRIPTION
**Pull request by mzj104**
修改内容
------------ 
scatter scatter_nd scatter_nd_add 没有对应关系，需要使用编写rules的方法：

```
paddle.scatter(x, index, updates, overwrite=True, name=None)
paddle.scatter_nd(index, updates, shape, name=None)
paddle.scatter_nd_add(x, index, updates, name=None)
```

需要特别注意的是，scatter函数在某些特定参数下，在gpu上运行时会出现随机的结果，导致accuracy出错。
详见：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/scatter_cn.html

为了避免这种随机性，应该要在overwrite == True 的情况下，尽量保证index不重复， 所以修改了config_analyzer.py的代码。当index的长度大于范围长度时，不可避免将产生重复值，配置中只有一个配置时这种不可避免的情况， 这个配置太大直接会oom， 已经放到了fresh report：
```
paddle.scatter(Tensor([183826, 256],"float32"), Tensor([10759233],"int64"), Tensor([10759233, 256],"float32"), )
```
----------
frexp 和 floor_mod使用mapping

paddle.frexp 和 torch.frexp对应：
```
paddle.frexp(x, name=None)
torch.frexp(input, out=None)
```
paddle.floor_mod虽然没有直接的对应关系，但经过验证paddle.floor_mod和paddle.mod完全一致，所以使用torch.fmod对应：
```
paddle.floor_mod(x, y, name=None)
torch.fmod(input, other, *, out=None)
```

测试的配置中，除前述的特例外，均能通过。

image

### CODE REVIEW

File: tester/paddle_to_torch/rules.py [hunk 1]
```diff
@@ -428,6 +428,62 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+class ScatterRule(BaseRule):
+    def apply(self, paddle_api: str) -> ConvertResult:
+        impl = """
+overwrite = locals().get('overwrite', True)
+x = x.clone()
+index = index.view(-1, 1)
+try:
+    updates = updates.expand_as(x)
+except:
+    pass
```
Comment (Cutelemon6) [inline tester/paddle_to_torch/rules.py:440]: 转换代码为什么需要错误处理？

File: tester/paddle_to_torch/rules.py [hunk 2]
```diff
@@ -428,6 +428,62 @@ def apply(self, paddle_api: str) -> ConvertResult:
...
+if not overwrite:
+    for i in range(index.shape[0]):
+        x[index[i]] = torch.zeros_like(x[index[i]])
+    for i in range(index.shape[0]):
+        x[index[i]] += updates[i]
+else:
+    for i in range(index.shape[0]):
+        x[index[i]] = updates[i]
```
Comment (Cutelemon6) [inline tester/paddle_to_torch/rules.py:448]: 如果 expand_as 抛出错误，就没有 updates 变量了，先解决上面的错误（消灭 try-catch，消灭后可能不用修改）

File: report/fresh_report/accuracy/test_log_paddle.scatter.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+paddle.scatter(Tensor([183826, 256],"float32"), Tensor([10759233],"int64"), Tensor([10759233, 256],"float32"), )
```
Comment (Cutelemon6) [inline report/fresh_report/accuracy/test_log_paddle.scatter.txt:1]: fresh_report 中放的是相关错误报告，这里没看到报错信息

**Review by Cutelemon6 — CHANGES_REQUESTED** [review]

File: report/fresh_report/accuracy/test_log_paddle.scatter.txt (hunk 3 above)
Comment (cangtianhuang) [reply report/fresh_report/accuracy/test_log_paddle.scatter.txt:1]: fresh report可以放报错配置+报错信息

Comment (mzj104) [reply report/fresh_report/accuracy/test_log_paddle.scatter.txt]: 已修改

**Review by wanghuancoder — APPROVED** [review]
LGTM
